typespec-extension/src/main/java/com/azure/autorest/TypeSpecPlugin.java

```diff
@@ -137,6 +137,8 @@ public String handlePartialUpdate(String filePath, String generatedContent) {
         SETTINGS_MAP.put("generic-response-type", true);
         SETTINGS_MAP.put("output-model-immutable", true);
         SETTINGS_MAP.put("disable-required-property-annotation", true);
+        // Defaulting to KeyCredential and not providing TypeSpec services to generate with AzureKeyCredential.
+        SETTINGS_MAP.put("use-key-credential", true);
```

Comment:
Would we set this as default for DPG as well? I can do in a follow-up PR.

For those already GAed, we can update their swagger/readme.md to still use AzureKeyCredential.

Reply:
Ideally, we would want to do this for all libraries (including DPG). This will be a 2-step process:
- update autorest to make this the default and release
- update readme.md of all currently GA'd libraries 

We can start with just DPG in a separate PR.